codex-rs/core/src/codex.rs

```diff
@@ -1026,12 +1025,14 @@ async fn run_turn(
             Err(CodexErr::Interrupted) => return Err(CodexErr::Interrupted),
             Err(CodexErr::EnvVar(var)) => return Err(CodexErr::EnvVar(var)),
             Err(e) => {
-                if retries < *OPENAI_STREAM_MAX_RETRIES {
+                // Use the configured provider-specific stream retry budget.
+                let max_retries = sess.client.get_provider().stream_max_retries();
+                if retries < max_retries {
                     retries += 1;
                     let delay = backoff(retries);
                     warn!(
                         "stream disconnected - retrying turn ({retries}/{} in {delay:?})...",
-                        *OPENAI_STREAM_MAX_RETRIES
+                        max_retries
```

Comment:
GitHub makes it hard to show this correctly...

```suggestion
                        "stream disconnected - retrying turn ({retries}/{max_retries} in {delay:?})...",
```

I thought the latest `clippy` flags this sort of thing, I should investigate...